common/app/routes/challenges/utils.js

```diff
@@ -488,3 +488,34 @@ export function unfilterMapUi(tree) {
     }
   );
 }
+
+export function unlockStep(challengeId, index) {
+  if (!challengeId || !index) {
+    return false;
+  }
+  if (typeof Storage !== 'undefined') {
```

Comment:
These utils should not have side effects. 

Side effects should happen in an epic.